nilearn/interfaces/fmriprep/load_confounds_compcor.py

```diff
@@ -42,22 +44,17 @@ def _select_compcor(compcor_cols, n_compcor):
     # only select if not "all", or less components are requested than there
     # actually is
     if (n_compcor != "all") and (n_compcor < len(compcor_cols)):
-        compcor_cols = compcor_cols[0:n_compcor]
+        compcor_cols = compcor_cols[:n_compcor]
```

Comment:
@htwangtw would it make sense to add a check around here? Something like a warning if no CompCor regressors are found?

Reply:
Good point - I have yet encountered this case but is it possible with how compcor is done?
I know it's possible for fMRIPrep to output confounds with no `cosineXX` regresssors

Reply:
I hadn't really considered that. I don't know if fMRIPrep would ever not write out compcor regressors, short of a failure within the workflow. I was more thinking that, if fMRIPrep changes things again, having warnings in these functions would make it easier for users to tell something's gone wrong.

Reply:
Right now the behaviour is the function will just fail, and a check is implemented in `_load_single_confounds_file`, so we don't need to write things in each function

https://github.com/nilearn/nilearn/blob/ec57e566455b3d929480fc2ce19e151e16673c6e/nilearn/interfaces/fmriprep/load_confounds.py#L315

The only variable that could be missing when fMRIPrep can run correctly is `cosineXX`, when the timeseries is too short. 

This deserves it's own issue

Reply:
Sorry for the delay in responding. I don't think `_check_error` catches the missing CompCor regressors. I tried running `load_confounds` on an fMRIPrepped file from 22.0.0 with the following kwargs:

```
{'strategy': ['motion', 'high_pass', 'compcor'],
 'motion': 'derivatives',
 'compcor': 'anat_separated',
 'n_compcor': 5}
```

It successfully ran, but the only regressors it returned were the cosine terms, motion parameters, and the motion derivatives. I wouldn't expect 22.x acompcor confounds to be loaded correctly per #3268, but I would have expected an error.

Reply:
No worries! I really appreciate that you took time to look into this!
I will have more time to work on this next week..

Reply:
checked on a 23.0.2 dataset

```python
from nilearn.interfaces.fmriprep import load_confounds

nii = "/home/remi/datalad/datasets.datalad.org/openneuro-derivatives/ds003768-fmriprep/sub-01/func/sub-01_task-rest_run-1_desc-preproc_bold.nii.gz"

confounds_out, sample_mask_out = load_confounds(
    nii,
    strategy=["motion", "high_pass", "compcor"],
    motion="derivatives",
    compcor="anat_separated",
    n_compcor=5,
)

print(sorted(confounds_out.columns))
```

and got this:

```python
['c_comp_cor_00', 'c_comp_cor_01', 'c_comp_cor_02', 'c_comp_cor_03', 'c_comp_cor_04', 'cosine00', 'cosine01', 'cosine02', 'cosine03', 'cosine04', 'cosine05', 'cosine06', 'cosine07', 'rot_x', 'rot_x_derivative1', 'rot_y', 'rot_y_derivative1', 'rot_z', 'rot_z_derivative1', 'trans_x', 'trans_x_derivative1', 'trans_y', 'trans_y_derivative1', 'trans_z', 'trans_z_derivative1', 'w_comp_cor_00', 'w_comp_cor_01', 'w_comp_cor_02', 'w_comp_cor_03', 'w_comp_cor_04']
```

AFAICT this test also catches missing regressors:

https://github.com/nilearn/nilearn/blob/68cfa06023edd4111aea681b6a157cc1cb2ac5e0/nilearn/interfaces/fmriprep/tests/test_load_confounds_components.py#L30